api/handlers.go

```diff
@@ -228,6 +239,36 @@ func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ h
 	api.WriteJSON(w, response)
 }
 
+// userUploadDeleteHandler unpins a single upload by this user.
+func (api *API) userUploadDeleteHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub, _, _, err := jwt.TokenFromContext(req.Context())
+	if err != nil {
+		api.WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, false)
+	if errors.Contains(err, database.ErrUserNotFound) {
+		api.WriteError(w, err, http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	uid := ps.ByName("uploadId")
+	_, err = api.staticDB.UnpinUpload(req.Context(), uid, *u)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	api.WriteSuccess(w)
+	// Now that we've returned results to the caller, we can take care of some
+	// administrative details, such as user's quotas check.
+	// Nota that this call is not affected by the request's context, so we use
+	// a separate one.
+	api.checkUserQuotas(context.Background(), u)
```

Comment:
Doesn't it make more sense to just fire up a goroutine that handles this? It's a little bit awkward to be doing stuff after you've written success. Also I'm  not sure whether the connection isn't kept open for as long as this function didn't return... Isn't it possible this is keeping alive the client connection, even though we've already written the response headers and status code? Not sure.

Reply:
TBH, I'm not sure either. My initial implementation used a goroutine for exactly those reasons but then I decided that it's only complicating things. I'll switch to a goroutine in a F/U, it's a single `go` keyword anyway.